### PR TITLE
Adding SSL Support

### PR DESCRIPTION
**Pull request by Matt-Lane**
This pull request adds SSL support to the configuration with this option:
**ssl**: _boolean_, default false

Usage example:
```
input {
   redis {
   	host => "securehost-redis.example.org"
   	port => 6380
   	data_type => "list"
   	key => "keyname"
	ssl => true
   }
}
```

Thanks for contributing to Logstash! If you haven't already signed our CLA, here's a handy link: https://www.elastic.co/contributor-agreement/

### CODE REVIEW

**Comment by jordansissel** [comment]
Thank you for working to improve logstash!

I wonder about the usefulness of this. SSL is great, but it needs correct
configuration (certificates, etc) that this PR does not enable.

Thoughts?

**Comment by Matt-Lane** [comment]
Looks like while my pull request worked for my specific situation (connecting to a secure redis cache in Azure), without setting any additional params, that will probably not be true for everyone. 
I can see in the docs:[https://github.com/redis/redis-rb#ssltls-support](https://github.com/redis/redis-rb#ssltls-support) that we'd need to pass in a hash of all the params. Would we need to individually verify each of the possible inputs to the [SSLContext](http://ruby-doc.org/stdlib-2.3.0/libdoc/openssl/rdoc/OpenSSL/SSL/SSLContext.html), or just accept a hash?
A note that I know exactly enough Ruby to have made this change, but not much more, so updates may be slow.

**Comment by jordansissel** [comment]
Thank you for the info! In this case, I think we could merge just what you
have. We will review shortly and help advise on any next steps.

**Comment by iremmats** [comment]
We are very interested in using this in production soon. Today we have to use stunnel to connect to azure redis with logstash.

**Comment by jstoja** [comment]
@jordansissel Any update on this?

**Comment by sstarcher** [comment]
Anything we can do to get this merged?

**Comment by CarlosSardo** [comment]
Hi! Can we have this PR merged? According to travis-ci logs, build [#73.4](https://travis-ci.org/logstash-plugins/logstash-input-redis/builds/287289649?utm_source=github_status&utm_medium=notification) passed!

**Comment by jstoja** [comment]
Maybe someone else from Elastic might be interested about this, perhaps @jsvd

File: lib/logstash/inputs/redis.rb [hunk 1]
```diff
@@ -29,6 +29,9 @@ module LogStash module Inputs class Redis < LogStash::Inputs::Threadable
   # The port to connect on.
   config :port, :validate => :number, :default => 6379
 
+  # SSL
+  config :ssl, :validate => :boolean, :default => false
```
Comment (jsvd) [inline lib/logstash/inputs/redis.rb]: this must be documented in the `docs/index.asciidoc` file, as that is the source of truth for the plugin's documentation

**Comment by sstarcher** [comment]
@jsvd would it be acceptable to get this in without going the extra mile.  As is it is already useful to people.  Certainly the extra features could be added later.

@Matt-Lane any chance you would have time to update the documentation

**Comment by jsvd** [comment]
I'm ok with not requiring other ssl settings, but it must have docs and at least a test that checks it's passing the right option to the redis library...

**Comment by sstarcher** [comment]
@jsvd fully agreed
@Matt-Lane If you don't think you will have time in the next week I can pick this up for you and add the docs and rebase it.  Please let me know.

File: lib/logstash/inputs/redis.rb (hunk 1 above)
Comment (Matt-Lane) [reply lib/logstash/inputs/redis.rb]: Added the requested documentation

**Comment by Matt-Lane** [comment]
I added the documentation and I believe I've rebased this correctly. Please let me know if I did not, or if any other changes should be made.

File: lib/logstash/inputs/redis.rb (hunk 1 above)
Comment (Yariv-Hashai) [reply lib/logstash/inputs/redis.rb]: @jsvd Can we merge it?

**Comment by robinjoseph08** [comment]
@jsvd Any chance you could take another look? We'd love to be able to use this plugin instead of forking it to get this functionality. Thanks!

**Pull request merged**